pkgs/by-name/so/soteria/package.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  wrapGAppsHook4,
+  cairo,
+  gdk-pixbuf,
+  glib,
+  gtk4,
+  pango,
+  polkit,
+}:
+rustPlatform.buildRustPackage {
+  pname = "soteria";
+  version = "0-unstable-2024-07-21";
+
+  src = fetchFromGitHub {
+    owner = "ImVaskel";
+    repo = "soteria";
+    rev = "898b7beab68a5e935f501b32de4f92bc091fa806";
+    hash = "sha256-lhS+37DqSgZrgrYgKGUpKMC22Qjdq9LPNS5k/dqvkRY=";
+  };
+
+  cargoHash = "sha256-kxDWU+HpnIzaOBsZwF363lSbipfmPdBoeg+spXDKAM4=";
+
+  nativeBuildInputs = [
+    pkg-config
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    cairo
+    gdk-pixbuf
+    glib
+    gtk4
+    pango
+  ];
+
+  # From upstream packaging:
+  #  Takes advantage of nixpkgs manually editing PACKAGE_PREFIX by grabbing it from
+  #  the binary itself.
+  #  https://github.com/NixOS/nixpkgs/blob/9b5328b7f761a7bbdc0e332ac4cf076a3eedb89b/pkgs/development/libraries/polkit/default.nix#L142
+  #  https://github.com/polkit-org/polkit/blob/d89c3604e2a86f4904566896c89e1e6b037a6f50/src/polkitagent/polkitagentsession.c#L599
+  preBuild = ''
+    export POLKIT_AGENT_HELPER_PATH="$(strings ${polkit.out}/lib/libpolkit-agent-1.so | grep "polkit-agent-helper-1")"
+  '';
+
+  meta = {
+    description = "A Polkit authentication agent written in GTK designed to be used with any desktop environment.";
```

Comment:
```suggestion
    description = "Polkit authentication agent written in GTK designed to be used with any desktop environment";
```

Reply:
Ah right, the derivation was written before the indefinite article rule was added.

I'll fix it shortly.